src/System.Private.CoreLib/src/Internal/Runtime/Augments/RuntimeThread.Unix.cs

```diff
@@ -2,62 +2,134 @@
 
-        private void StartCore(object parameter)
+        private bool SetPriorityLive(ThreadPriority priority)
         {
-            // TODO: Start a new thread using _threadStart and parameter
+            return true;
         }
 
-        public ApartmentState GetApartmentState()
+        [NativeCallable]
+        private static void OnThreadExit()
         {
-            Environment.FailFast(
-                "Should not reach here because CoreFX's Thread class should not forward to this function on Unix");
-            return ApartmentState.Unknown;
+            // Set the Stopped bit and signal the current thread as stopped
+            RuntimeThread currentThread = t_currentThread;
+            if (currentThread != null)
+            {
+                int state = currentThread._threadState;
+                if ((state & (int)(ThreadState.Stopped | ThreadState.Aborted)) == 0)
+                {
+                    currentThread.SetThreadStateBit(ThreadState.Stopped);
+                }
+                currentThread._stopped.Set();
+            }
         }
 
-        public bool TrySetApartmentState(ApartmentState state)
+        private ThreadState GetThreadState() => (ThreadState)_threadState;
+
+        private bool JoinInternal(int millisecondsTimeout)
         {
-            Environment.FailFast(
-                "Should not reach here because CoreFX's Thread class should not forward to this function on Unix");
-            return false;
+            // This method assumes the thread has been started
+            Debug.Assert(!GetThreadStateBit(ThreadState.Unstarted) || (millisecondsTimeout == 0));
+            SafeWaitHandle waitHandle = _stopped.SafeWaitHandle;
+
+            // If an OS thread is terminated and its Thread object is resurrected, waitHandle may be finalized and closed
+            if (waitHandle.IsClosed)
+            {
+                return true;
+            }
+
+            // Prevent race condition with the finalizer
+            try
+            {
+                waitHandle.DangerousAddRef();
+            }
+            catch (ObjectDisposedException)
+            {
+                return true;
+            }
+
+            try
+            {
+                return _stopped.WaitOne(millisecondsTimeout);
+            }
+            finally
+            {
+                waitHandle.DangerousRelease();
+            }
         }
 
-        public void DisableComObjectEagerCleanup() { }
+        private bool CreateThread(GCHandle thisThreadHandle)
+        {
+            // Avoid OOM after creating the thread
+            var stopped = new ManualResetEvent(false);
+
+            if (!Interop.Sys.RuntimeThread_CreateThread((IntPtr)_maxStackSize,
+                AddrofIntrinsics.AddrOf<Interop.Sys.ThreadProc>(StartThread), (IntPtr)thisThreadHandle))
+            {
+                return false;
+            }
+
+            // This marks the thread as being started
+            _stopped = stopped;
```

Comment:
_stopped is initialized only once, those are two exclusive code paths: foreign threads cannot be started.